### ISSUE TITLE
Simple Window Switcher is broken on Windows 11 24H2

### ISSUE THREAD

**Issue by andrewz1986** [valinet/ExplorerPatcher#3765]
I've been using Simple Window Switcher for years and have not noticed any issues. After updating Windows 11 to version 24H2, there is noticeable delay when switching windows, although not dramatic. The delay is not caused by "Show delay" option in the ExplorerPatcher settings. 

I also noticed that using Alt + Tab twice in consecution does not bring the original app in the foreground – tab switching is moving forward only. That is a deviation from the way original window switcher works.

Can somebody fix please? Thanks.

**Comment by pyrates999** [comment]
is it this? https://github.com/valinet/ExplorerPatcher/issues/3720

If so, please close this.

**Comment by andrewz1986** [comment]
No. I'm not talking about Windows Switcher Style: Windows NT. I'm talking about Windows Switcher Style: Simple Window Switcher. That's the one that allows detailed settings in ExplorerPatcher regarding its UI and behavior. Thanks.

**Comment by kjellerstedt** [comment]
Same issue here, for me the Simple Window Switcher does not even show up or  register any alt-tab:ing. Windows 11 24H2

**Comment by Amrsatrio** [comment]
I will check why it is stuck in an infinite loop this week. RIght now on 24H2 bug fixing spree.

**Comment by kjellerstedt** [comment]
Any  updates on this?

**Comment by Amrsatrio** [comment]
Not yet, apparently I had other stuff to do and backlogged this one... I'll let you know once I've figured this out.

**Comment by Alexey-Ivanensky** [comment]
+1

**Comment by musaAmir** [comment]
I am still experiencing this issue with the Simple Window Swither after updating to 24H2 a while back. Alt tab doesn't work properly. 

Whenever I alt + tab it selects the last window in the list instead of the default which is the second item (last opened window). Also, messing with the settings seem to break the entire Window Switcher till it doesn't function at all and a PC restart is required.

**Comment by Hyperi0n1337** [comment]
Just wanted to report that I also have this issue with the Simple Windows Switcher. Regular Alt + Tab works fine, its alt + tilde for cycling through the windows of a specific program where the first item to be selected is the last one from the group (check the screenshot to see what I mean, this is what's selected once alt tilde is pressed once) 
![image](https://github.com/user-attachments/assets/301f6d93-51a1-4c05-982b-73b7039a180f)

**Comment by Demitrix2k** [comment]
> I am still experiencing this issue with the Simple Window Swither after updating to 24H2 a while back. Alt tab doesn't work properly.
> 
> Whenever I alt + tab it selects the last window in the list instead of the default which is the second item (last opened window). Also, messing with the settings seem to break the entire Window Switcher till it doesn't function at all and a PC restart is required.
Experiencing the very same after upgrading to 24H2

**Comment by philipp** [comment]
Seems like this is the placeholder for all of the different misbehaviors of SWS in 24H2?
If so, I found another duplicate while searching for the behavior I was seeing: 
#3626 describes the "Alt-Tab moves to last item instead of next item" behavior
Along with that, I also saw that turning on "show the desktop as the last window in the interaction list" completely disables Alt-Tab function until the setting is turned off again and File Explorer is restarted

**Comment by black-ish** [comment]
I also noticed the slowness suddenly on a new Win 11 installation with the newest Update integrated (26100.3476)
It wasn't there with a slightly older update version, somewhere in the 2xxx range, I can check if it's of interest.
Both use Explorer Patcher 22621.4317.67.1 and same settings and same software all around even the same versions of the software.

What I noticed was interesting, if you happen to install Explorer Patcher the first time and switch the Window Switcher the first time to "Simple Window Switcher" and press Alt+Tab the first time, it's fast amazingly so, but all consecutive times it takes about 800ms+ to open.

**Comment by Amrsatrio** [comment]
The fix for SWS not showing when you have chosen the wallpaper to be visible in the list is coming in the next release.

**Comment by oneofmanyacdcfanstoo** [comment]
I too am experiencing teh problum with Simple Window Switcher. Fortunately, mine is where it just doesn't do anything. NT Switcher and all the others work fine.

**Comment by Amrsatrio** [comment]
Fixed in 68.2.

**Comment by andrewz1986** [comment]
Thanks for releasing new version of EP and taking care of Simple Window Switcher. Unfortunately, it does not work as good as it did before 24H2. I'm specifically talking about briefly pressing Alt+Tab to switch two windows back and forth. Important note: I have this keyboard combo assigned to my mouse via Logitech Options and exclusively switch windows by tilting mouse scrolling wheel to the left. That makes the Alt+Tab action input even more brief.

First thing is that there is still certain lag in comparison with regular Windows 11 window switcher. The worse thing is that it does not work in like 30 percent of cases. These fails may be result of the lag, since pressing Alt+Tab on the keyboard (since I have to physically press two keys, the action input is not as brief) works mostly well. 

Please check my video. Everytime the on-screen keyboard shows that Alt was pressed, I was actually tilting my mouse scrolling wheel. https://www.youtube.com/watch?v=eeLwiqDCc38

**Comment by pyrates999** [comment]
Try changing Show Delay to a lower value in ms.

**Comment by andrewz1986** [comment]
> Try changing Show Delay to a lower value in ms. It is set to 150 ms in your video.

Show delay relates to window switching UI (Alt+Tab menu). My issue has nothing to do with that. I am talking about reliability and swiftness of two windows switching back and forth.

**Comment by black-ish** [comment]
I'd like to add here that for me nothing changed in regards to sws with the newest update.
There is still a fat delay for it to show.
Only output in debug window:
```
[sws] WindowSwitcher::Show 10148 [[ 927 + 0 + 0 + 2 = 929 ]]
[sws] CalculateHelper 0 [[ 2 + 2 = 4 ]].
```
My delay is set to none in settings.

**Comment by andrewz1986** [comment]
I'm not sure what happened, whether Microsoft made some update to Windows 11 (I have build 26100.4484), or some symbols were downloaded to Explorer Patcher, but Simple Windows Switcher now seems to work very well again. Hopefully it will stay like that. Thanks. These are my settings. 

![Image](https://github.com/user-attachments/assets/c50f0178-c1e3-4126-8968-4b4bb718ddfb)

**Comment by am1re** [comment]
> [sws] WindowSwitcher::Show 10148 [[ 927 + 0 + 0 + 2 = 929 ]]
> [sws] CalculateHelper 0 [[ 2 + 2 = 4 ]].

same for me, update didnt fix delay for alt `

**Comment by Louis047** [comment]
I would like to reopen this issue but I can't lol. 

Anyways, I have installed the latest stable release of ExplorerPatcher and tested SWS with different `Show Delay` values and found that when setting below 100ms, there's still a noticeable delay and behaves as if a higher delay was set. It takes time to load the alt+tab area as well and I'm not sure what went wrong. 

Currently using Windows 11 24H2 Build 26100.7840

Below is the showcase video, you can notice that when trying to alt tab rapidly, you can briefly feel/notice the delay thereby indicating that the option isn't respected at all:

https://github.com/user-attachments/assets/a16faec6-8685-4a8f-be7a-c6728b2e6802

**Comment by ruslan-mogilevskiy** [comment]
This delay (for me is ~1sec) after alt-tab is really annoying. It affects not only how instantly the window selector appears, but how instantly windows are switching

**Comment by andrewz1986** [comment]
> This delay (for me is ~1sec) after alt-tab is really annoying. It affects not only how instantly the window selector appears, but how instantly windows are switching

I'm running Windows 11 25H2 build 26200.8117 and ExplorerPatcher 26100.4946.69.6 and I do not observe any notable issues regarding Simple Windows Switcher delay. The settings allow to change the value from 0 to 500 ms. I set it to 75 ms. When Alt-Tabbing, windows switching is almost instant, only slightly slower that Windows 11 default windows switcher. So yes, there is room for improvements, but the issue is certainly not as serious in my case as you describe. What versions are you running?

**Comment by Louis047** [comment]
> I would like to reopen this issue but I can't lol.
> 
> Anyways, I have installed the latest stable release of ExplorerPatcher and tested SWS with different `Show Delay` values and found that when setting below 100ms, there's still a noticeable delay and behaves as if a higher delay was set. It takes time to load the alt+tab area as well and I'm not sure what went wrong.
> 
> Currently using Windows 11 24H2 Build 26100.7840
> 
> Below is the showcase video, you can notice that when trying to alt tab rapidly, you can briefly feel/notice the delay thereby indicating that the option isn't respected at all:
> 2026-02-26.21-46-02.mp4

@andrewz1986 hey there, check this out as well. Do you use windhawk? I highly doubt that it might be the cause as it injects directly into some critical windows processes (like `dwm.exe` etc), at least for the mods I use.

**Comment by pyrates999** [comment]
> This delay (for me is ~1sec) after alt-tab is really annoying. It affects not only how instantly the window selector appears, but how instantly windows are switching

If you have wallpaper engine installed, uninstall it.  It is resource intensive.

**Comment by Louis047** [comment]
Hmm, this is odd. the delay issue fixed for me automatically. I think it was after I reinstalled ExplorerPatcher after a few weeks. Well, at least it fixed itself. 

Windows 11 24H2 Build 26100.8117
